docs/tutorial/beginning_decomp.md

```diff
@@ -0,0 +1,691 @@
+```
+$ ./tools/m2ctx.py <path_to_c_file>
+```
+from the main directory of the repository. In this case, the C file is `src/overlays/actors/ovl_En_Recepgirl/z_en_recepgirl.c`. This generates a file called `ctx.c` in the main directory of the repository.
+
+To get mips_to_c to decompile a function, the bare minimum is to run
+```
+$ ../mips_to_c/mips_to_c.py <path_to_function_assembly_file>
+```
+(from the root directory of `mm`). We can tell mips2c to use the context file we just generated by adding `--context ctx.c`. If we have data, mips2c may be able to assist with that as well.
+
+In this case, we want the assembly file for `EnRecepgirl_Init`. You can copy the path to the file in VSCode or similar, or just tab-complete it once you know the directory structure well enough: it turns out to be `asm/non_matchings/overlays/ovl_En_Recepgirl/EnRecepgirl_Init.s`.
+
+**N.B.** You want the file in `nonmatchings`! the files in the other directories in `asm/` are the *unsplit* asm, which can be used, but is less convenient (you would need to include the rodata, for example, and it will do the whole file at once. This is sometimes useful, but we'll go one function at a time today to keep things simple).
+
+We shall also include the data file, which is located at `data/overlays/ovl_En_Recepgirl/ovl_En_Recepgirl.data.s`. Hence the whole command will be
+```
+$ ../mips_to_c/mips_to_c.py asm/non_matchings/overlays/ovl_En_Recepgirl/EnRecepgirl_Init.s data/ovl_En_Recepgirl/ovl_En_Recepgirl.data.s --context ctx.c
+```
+
+However, there is one more thing we should do before we run this. *For the main four functions only*, a very common pattern is that the first argument is recast to a new variable with the actual actor's type. This allows us to access the extra struct variables that this particular actor has, rather than just everything in the general `Actor` struct. More simply, we will end up with
+```C
+EnRecepgirl* this = THIS;
+```
+at the top of the function, where you can see from the top of the file that `THIS` just expands to `(EnRecepgirl*)thisx`. Since `Init` (and the other "main four" functions) will access these struct variables, it is a good idea to change the `Init` prototype to be wrong, to fool mips2c so that it will output the actual variable rather than just its offset in the struct.
+
+That's a lot of words for: change `Actor* thisx` in the `Init` prototype in the C file to `EnRecepgirl* this`:
+```C
+void EnRecepgirl_Init(EnRecepgirl *this, GlobalContext *globalCtx)
+```
+and remake the context file. (It's a good idea to do the same for all of the "main four", so we won't mention it again). *Once you have the function decompiled, remember to put the arguments back to how they should be in both the prototype and the function itself*
```

Comment:
This shouldn't be required anymore. The latest mips2c can detect this pattern and do it automatically.

Specifically, if:
- The function name starts with `Foo_`
- The first argument has type `Ptr *`
- There exists a struct called `Foo` in the context, whose first member is of type `Ptr`

Then, mips2c it automatically downcasts the first argument from a `Ptr *` into a `Foo *` & creates a `this`/`self` variable.